api/src/main/java/ai/djl/ndarray/NDArray.java

```diff
@@ -157,23 +157,14 @@ default boolean isScalar() {
      */
     NDArray toType(DataType dataType, boolean copy);
 
-    /**
-     * Attaches a gradient {@code NDArray} to this {@code NDArray} and marks it so {@link
-     * ai.djl.training.GradientCollector#backward(NDArray)} can compute the gradient with respect to
-     * it.
-     */
-    default void requiresGradient() {
-        requiresGradient(true);
-    }
-
     /**
      * Attaches a gradient {@code NDArray} to this {@code NDArray} and marks it so {@link
      * ai.djl.training.GradientCollector#backward(NDArray)} can compute the gradient with respect to
      * it.
      *
      * @param requiresGrad if {@code NDArray} requires gradient or not
      */
-    void requiresGradient(boolean requiresGrad);
+    void setRequiresGradient(boolean requiresGrad);
```

Comment:
do we still want a default setRequirersGradient()? do that sound like Java-friend API?

Reply:
It's a bit confusing here, even we what to keep it, it should be requireGradient()